lib/src/mock_document_snapshot.dart

```diff
@@ -10,10 +11,18 @@ class MockDocumentSnapshot extends Mock implements DocumentSnapshot {
   final Map<String, dynamic> _document;
   final bool _exists;
   final MockDocumentReference _reference;
+  MockSnapshotMetadata _metadata;
 
   MockDocumentSnapshot(
-      this._reference, this._id, Map<String, dynamic> document, this._exists)
-      : _document = deepCopy(document);
+      this._reference, this._id, Map<String, dynamic> document, this._exists,
+      {MockSnapshotMetadata metadata})
```

Comment:
Since none of the code currently passes metadata, would you mind simply initializing it like so: `_metadata = MockSnapshotMetadata()` at line 14?

I'd prefer to keep the code as simple as possible. If in the future, you plan to make a system to set up custom metadata, then we can make the constructor take the optional parameter in another PR.

https://github.com/atn832/cloud_firestore_mocks/search?q=MockDocumentSnapshot

Reply:
Ok, I did this like you suggested. Thanks